File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
    bool fun(int i,vector<int>&nums){
        if(i==nums.size()-1)return true;
        if(i>=nums.size())return false;
        for(int j=1;j<=nums[i];j++){
            if(fun(i+j,nums))return true;
        }
        return false;
    }
public:
    bool canJump(vector<int>& nums) {
        return fun(0,nums);
    }
};

File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        n=len(nums)

        def fun(i,cur):
            if(i==n and cur==0):
                return 1

            if(i==n):
                return 0

            pos=fun(i+1,cur-nums[i])
            neg=fun(i+1,cur+nums[i])

            return pos+neg


        return fun(0,target)